fe1-web/src/core/network/ingestion/__tests__/IngestionHandler.test.ts

```diff
@@ -0,0 +1,67 @@
+import { mockKeyPair, mockMessageRegistry } from '__tests__/utils';
+import {
+  Broadcast,
+  ExtendedJsonRpcRequest,
+  JsonRpcMethod,
+  JsonRpcRequest,
+} from 'core/network/jsonrpc';
+import {
+  ActionType,
+  configureMessages,
+  Message,
+  MessageData,
+  ObjectType,
+} from 'core/network/jsonrpc/messages';
+import { dispatch } from 'core/redux';
+
+import { ExtendedMessage } from '../ExtendedMessage';
+import { handleExtendedRpcRequests } from '../Handler';
+import { addMessages } from '../MessageReducer';
+
+// region mocks
+
+jest.mock('core/redux', () => {
+  const actualModule = jest.requireActual('core/redux');
+  return {
+    ...actualModule,
+    dispatch: jest.fn().mockImplementation(() => {}),
+  };
+});
+
+const mockAddress = 'some address';
+const mockChannel = 'some channel';
+const mockMessageData: MessageData = { object: ObjectType.ELECTION, action: ActionType.OPEN };
+
+// these can only be instantiated after mocking the message registry
+let mockMessage: Message;
+let mockJsonRequest: Partial<JsonRpcRequest>;
+let extendedRequest: ExtendedJsonRpcRequest;
+
+// endregion
+
+beforeAll(() => {
+  configureMessages(mockMessageRegistry);
+  mockMessage = Message.fromData(mockMessageData, mockKeyPair);
+  mockJsonRequest = {
+    jsonrpc: 'some data',
+    method: JsonRpcMethod.BROADCAST,
+    params: { channel: mockChannel, message: mockMessage } as Broadcast,
+  };
+  extendedRequest = new ExtendedJsonRpcRequest({ receivedFrom: mockAddress }, mockJsonRequest);
+});
+
+describe('handleExtendedRpcRequests', () => {
+  it('dispatches the correct redux action', () => {
+    handleExtendedRpcRequests(extendedRequest);
+    const obj = addMessages(
+      ExtendedMessage.fromMessage(mockMessage, mockChannel, mockAddress).toState(),
+    );
+
+    // the receivedAt value can differ. calls[i][j] => jth argument of ith call
+    // @ts-ignore
+    delete obj.payload[0].receivedAt;
+    delete (dispatch as jest.Mock).mock.calls[0][0].payload[0].receivedAt;
+
+    expect(dispatch).toHaveBeenCalledWith(obj);
```

Comment:
While I could forgive the `@ts-ignore`, the intention behind your code is not very understandable, and that's a much bigger problem.

It looks like this is what you want to do:
https://stackoverflow.com/a/69291149/9533693
... and I think you'll agree this is much easier to understand as well 😄

Reply:
That seems to be working, thank you very much!! I tried something similar before and it didn't work but apparently I just messed up somewhere :)